Tests/UnitTests/EdgeEventWarningTests.swift

```diff
@@ -0,0 +1,160 @@
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+@testable import AEPEdge
+import XCTest
+
+class EdgeEventWarningTests: XCTestCase {
+
+    override func setUp() {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+        continueAfterFailure = false // fail so nil checks stop execution
+    }
+
+    func testCanDecode_eventWarning_allParams() {
+        // setup
+        let jsonData = """
+                        {
+                          "eventIndex": 1,
+                          "type": "https://ns.adobe.com/aep/errors/EXEG-0204-200",
+                          "status": 200,
+                          "title": "test title",
+                          "report": {
+                            "cause": {
+                              "message": "Cannot read related customer for device id: ...",
+                              "code": 202
+                            }
+                          }
+                        }
+                      """.data(using: .utf8)
+
+        // test
+        let warning = try? JSONDecoder().decode(EdgeEventWarning.self, from: jsonData ?? Data())
+
+        // verify
+        XCTAssertEqual(1, warning?.eventIndex)
+        XCTAssertEqual("https://ns.adobe.com/aep/errors/EXEG-0204-200", warning?.type)
+        XCTAssertEqual(200, warning?.status)
+        XCTAssertEqual("test title", warning?.title)
+        XCTAssertEqual("Cannot read related customer for device id: ...", warning?.report?.cause?.message)
+        XCTAssertEqual(202, warning?.report?.cause?.code)
+    }
+
+    func testCanDecode_eventWarning_allParams_multipleWarnings() {
+        // setup
+        let jsonData = """
+                        [
+                            {
+                              "eventIndex": 1,
+                              "type": "https://ns.adobe.com/aep/errors/EXEG-0204-200",
+                              "status": 200,
+                              "title": "test title",
+                              "report": {
+                                "cause": {
+                                  "message": "Cannot read related customer for device id: ...",
+                                  "code": 202
+                                }
+                              }
+                            },
+                            {
+                              "eventIndex": 2,
+                              "type": "https://ns.adobe.com/aep/errors/EXEG-0204-202",
+                              "status": 202,
+                              "title": "test title 2",
+                              "report": {
+                                "cause": {
+                                  "message": "Cannot read related customer for device id: ...",
+                                  "code": 202
```

Comment:
Can you also include some tests with unexpected parameters? We should capture the case where the server gets updated with new params, but the client runs current version.

Reply:
If I'm understanding this correctly, this isn't possible in Swift. We can only decode the properties/types we explicitly define in `EdgeEventWarning`. Anything that doesn't fit the definition in `EdgeEventWarning` will be discarded. If the format of Konductor warnings updates to append new properties we'll need to update the SDK to decode these.

Reply:
That is correct, I think it is fine to discard extraneous keys until we introduce support for them, I am just suggesting to have a test to make sure we don't crash / drop the response if the server response gets updated.